### PR TITLE
Feature/Startup config

### PR DESCRIPTION
**Pull request by zhxymh**

### CODE REVIEW

File: src/AElf.Blockchains.MainChain/MainChainAElfModule.cs [hunk 1]
```diff
@@ -46,6 +47,39 @@ public MainChainAElfModule()
 
         public override void OnApplicationInitialization(ApplicationInitializationContext context)
         {
+            var nodeType = context.ServiceProvider.GetService<IOptionsSnapshot<NodeOptions>>().Value.NodeType;
+            var tokenInitialOptions = context.ServiceProvider.GetService<IOptionsSnapshot<TokenInitialOptions>>().Value;
+            
+            var tokenSymbol=  tokenInitialOptions.Symbol;
```
[the remark this answers is not in the document]
Comment (zhxymh) [reply src/AElf.Blockchains.MainChain/MainChainAElfModule.cs]: You mean don't define variables，directly assign a value to tokenInitialOptions？

[the remark this answers is not in the document]
Comment (loning) [reply src/AElf.Blockchains.MainChain/MainChainAElfModule.cs]: yes

**Comment by loning** [comment]
we should add a native symbol in bridge context, if we store data in state, it will make our contact very slow.

state will be reloaded in every smart contract method call, or we should provide a readonly state cached in smart contract runtime. 

@gldeng, what's your opinion?

**Comment by gldeng** [comment]
> we should add a native symbol in bridge context, if we store data in state, it will make our contact very slow.
> 
> state will be reloaded in every smart contract method call, or we should provide a readonly state cached in smart contract runtime.
> 
> @gldeng, what's your opinion?

I agree. So the cached value will be read from Token contract right? Does the bridge context know the Token contract address? Or it can be a configuration for the chain like ChainId?

File: src/AElf.Contracts.CrossChain/CrossChainContract_Helper.cs [hunk 2]
```diff
@@ -70,7 +71,7 @@ private void LockTokenAndResource(SideChainCreationRequest sideChainInfo, int ch
                 From = Context.Sender,
                 To = Context.Self,
                 Amount = sideChainInfo.LockedTokenAmount,
-                Symbol = Context.NativeTokenSymbol
+                Symbol = Context.Variables[ContextVariableDictionary.NativeSymbolName],
```
Comment (loning) [inline src/AElf.Contracts.CrossChain/CrossChainContract_Helper.cs]: => Context.Variables.NativeSymbol

**Comment by loning** [comment]
PLS understand my commit and then make changes, https://github.com/AElfProject/AElf/pull/1557/commits/3725d714945f9fada6f09f3578744120a05e7f18#diff-f38d9efccf8b35a73997a0b071c62a50R119

**Comment by zhxymh** [comment]
> PLS understand my commit and then make changes, [3725d71#diff-f38d9efccf8b35a73997a0b071c62a50R119](https://github.com/AElfProject/AElf/commit/3725d714945f9fada6f09f3578744120a05e7f18#diff-f38d9efccf8b35a73997a0b071c62a50R119)

Sorry, my mistake!

**Pull request merged**